python/pyarrow/plasma.pyx

```diff
@@ -370,7 +370,7 @@ cdef class PlasmaClient:
                                          object_buffers[i].metadata_size))
         return result
 
-    def put(self, object value, ObjectID object_id=None,
+    def put(self, object value, ObjectID object_id=None, int memcopy_threads=6,
```

Comment:
Should this default to something more like the number of physical CPUs (which may vary from machine to machine)?

Reply:
Perhaps as an upper bound. In some cases I think that will be too high, e.g., on a 64 core machine I did a few experiments and didn't see benefits past 20 threads. Something like `min(NUM_CPUS, 16)` might make sense.